packages/@aws-cdk/aws-ec2/lib/user-data.ts

```diff
@@ -12,6 +13,37 @@ export interface LinuxUserDataOptions {
   readonly shebang?: string;
 }
 
+/**
+ * Options when downloading files from S3
+ */
+export interface S3DownloadAndExecuteOptions {
+
+  /**
+   * Name of the bucket to download from
+   */
+  readonly bucketName: string;
```

Comment:
Sure I can swap this to an IBucket.

However I disagree that the User Data should be responsible for granting read permissions for the bucket since the User Data can be created separately from the instance and then passed in later and the UserData has no knowledge of what it is being used for.